### PR TITLE
fix(APPL-220): Fix Newly Submitted Artwork Images Are Not Loaded

### PR DESCRIPTION
**Pull request by lordkiz**
This PR resolves [APPL-220] 

### Description

- Fixes missing Image in MyCollection list view following a submission of artwork.

#### Before

https://user-images.githubusercontent.com/18648835/207552243-1a3537f2-1100-4294-b81b-665605530c81.mp4

#### After

https://user-images.githubusercontent.com/18648835/207552326-77bd913f-b4f9-499b-9d9a-7773c9e0ed43.mov

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [x] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Fix Newly Submitted Artwork Images Are Not Loaded - kizito

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[APPL-220]: https://artsyproduct.atlassian.net/browse/APPL-220?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/MyCollection/Components/MyCollectionArtworkListItem.tsx [hunk 1]
```diff
@@ -1,25 +1,106 @@
...
 
-  const { artist, date, image, internalID, medium, mediumType, slug, title } = artwork
+  const { artist, date, image, internalID, medium, mediumType, slug, title, submissionId } = artwork
+
+  useEffect(() => {
+    handleImages()
+  }, [myCollectionIsRefreshing])
+
+  const handleImages = async () => {
+    const [localVanillaArtworkImages, localSubmissionArtworkImages] = await Promise.all([
```
Comment (MrSltun) [inline src/app/Scenes/MyCollection/Components/MyCollectionArtworkListItem.tsx:95]: 🍦 😄

File: src/app/Scenes/MyCollection/Components/MyCollectionArtworkListItem.tests.tsx [hunk 2]
```diff
@@ -4,7 +4,10 @@ import { MyCollectionArtworkListItemTestsQuery } from "__generated__/MyCollectio
 import { navigate } from "app/navigation/navigate"
 import { mockTrackEvent } from "app/tests/globallyMockedStuff"
 import { renderWithWrappers } from "app/tests/renderWithWrappers"
+import { LocalImage } from "app/utils/LocalImageStore"
+import * as LocalImageStore from "app/utils/LocalImageStore"
 import { graphql, QueryRenderer } from "react-relay"
+import { act } from "react-test-renderer"
```
Comment (MrSltun) [inline src/app/Scenes/MyCollection/Components/MyCollectionArtworkListItem.tests.tsx:10]: **Question**: why are we using `act` from `react-test-renderer` and not from React Native testing library? 🤔

**Review by MrSltun — APPROVED** [review]
LGTM! 🌟 I just left a small comment :)

File: src/app/Scenes/MyCollection/Components/MyCollectionArtworkListItem.tests.tsx (hunk 2 above)
Comment (lordkiz) [reply src/app/Scenes/MyCollection/Components/MyCollectionArtworkListItem.tests.tsx:10]: I didn't know there is a preference. It is all the same I suppose. Should we not be using stuffs from `react-test-renderer`?

Comment (lordkiz) [reply src/app/Scenes/MyCollection/Components/MyCollectionArtworkListItem.tests.tsx:10]: Updating to testing library. Thanks @MrSltun

**Comment by lordkiz** [comment]
#squashongreen

File: src/app/Scenes/MyCollection/Components/MyCollectionArtworkListItem.tests.tsx (hunk 2 above)
Comment (MrSltun) [reply src/app/Scenes/MyCollection/Components/MyCollectionArtworkListItem.tests.tsx]: I actually don't know 😅 that's why I asked haha

Comment (gkartalis) [reply src/app/Scenes/MyCollection/Components/MyCollectionArtworkListItem.tests.tsx]: It is actually the same thing, but usually for consistency we use the one from `@testing-library/react-native`. From what I remember `@testing-library/react-native` re-exports the act from `react-test-renderer`

File: src/app/Scenes/MyCollection/Components/MyCollectionArtworkListItem.tsx [hunk 3]
```diff
@@ -1,25 +1,106 @@
...
 import { Flex, NoArtworkIcon, Text, Touchable } from "palette"
+import { useEffect, useState } from "react"
+import { Image as RNImage } from "react-native"
 import { useFragment } from "react-relay"
 import { useTracking } from "react-tracking"
 import { graphql } from "relay-runtime"
 
 export const ARTWORK_LIST_IMAGE_SIZE = 80
 
+const ListItemImageView: React.FC<{
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/Components/MyCollectionArtworkListItem.tsx:21]: **suggestion:** What do you think about extracting this component into its own file? I think we do this for almost every component to keep the file size small

File: src/app/Scenes/MyCollection/Components/MyCollectionArtworkListItem.tsx (hunk 1 above)
Comment (olerichter00) [reply src/app/Scenes/MyCollection/Components/MyCollectionArtworkListItem.tsx:95]: 🍦🍦🍦

**Review by olerichter00 — APPROVED** [review]
Thanks for the fix 🙏 

I've just left a comment about moving the component into it's own file.

**Pull request merged**